ds3-autogen-java/src/test/java/com/spectralogic/ds3autogen/java/generators/typemodels/BaseTypeGenerator_Test.java

```diff
@@ -0,0 +1,383 @@
+                new Ds3Element("ElementName1", "com.spectralogic.test.ElementType1", "ElementComponentType1"),
+                new Ds3Element("ElementName2", "ElementType2", "com.spectralogic.test.ElementComponentType2"),
+                new Ds3Element("ElementName3", "ElementType3", null));
+
+        final Ds3Type ds3Type = new Ds3Type(
+                "TypeName",
+                null,
+                ds3Elements,
+                null);
+
+        final ImmutableList<String> result = generator.getAllImports(ds3Type);
+        assertThat(result.size(), is(4));
+        assertTrue(result.contains("java.util.List"));
+        assertTrue(result.contains("com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper"));
+        assertTrue(result.contains("com.spectralogic.ds3client.models.ElementType1"));
+        assertTrue(result.contains("com.spectralogic.ds3client.models.ElementComponentType2"));
+    }
+
+    @Test
+    public void getXmlTagFromAnnotation_CustomName_Test() {
+        final Ds3Annotation annotation = createCustomMarshaledNameAnnotation();
+        final String result = getXmlTagFromAnnotation(annotation);
+        assertThat(result, is("Error"));
+    }
+
+    @Test
+    public void getXmlTagFromAnnotation_SimpleName_Test() {
+        final Ds3Annotation annotation = createSimpleNameAnnotation();
+        final String result = getXmlTagFromAnnotation(annotation);
+        assertThat(result, is(""));
+    }
+
+    @Test
+    public void getXmlTagFromAnnotation_NonCustomName_Test() {
+        final Ds3Annotation annotation = createNonCustomMarshaedNameAnnotation();
+        final String result = getXmlTagFromAnnotation(annotation);
+        assertThat(result, is(""));
+    }
+
+    @Test
+    public void getXmlTagFromAllAnnotations_NullList_Test() {
+        final String result = getXmlTagFromAllAnnotations(null, "ElementName");
+        assertThat(result, is(""));
+    }
+
+    @Test
+    public void getXmlTagFromAllAnnotations_EmptyList_Test() {
+        final String result = getXmlTagFromAllAnnotations(ImmutableList.of(), "ElementName");
+        assertThat(result, is(""));
+    }
+
+    @Test
+    public void getXmlTagFromAllAnnotations_FullList_Test() {
+        final ImmutableList<Ds3Annotation> annotations = ImmutableList.of(
+                createSimpleNameAnnotation(),
+                createNonCustomMarshaedNameAnnotation(),
+                createCustomMarshaledNameAnnotation());
+        final String result = getXmlTagFromAllAnnotations(annotations, "ElementName");
+        assertThat(result, is("Error"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void getXmlTagFromAllAnnotations_Exception_Test() {
+        final ImmutableList<Ds3Annotation> annotations = ImmutableList.of(
+                createCustomMarshaledNameAnnotation(),
+                createCustomMarshaledNameAnnotation());
+        getXmlTagFromAllAnnotations(annotations, "ElementName");
```

Comment:
i would add a fail() after getXmlTagFromAllAnnotations jsut to make it clear that we are not supposed to succeed this function. just a suggestion